pkg/controller/config/agent.go

```diff
@@ -41,43 +41,47 @@ func (ca *Agent) Start(configLocation string) error {
 		return err
 	}
 	ca.Set(c)
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logrus.WithError(err).Error("Error when creating watcher.")
+	}
+	ca.watcher = watcher
 	go func() {
-		var lastModTime time.Time
-		// Rarely, if two changes happen in the same second, mtime will
-		// be the same for the second change, and an mtime-based check would
-		// fail. Reload periodically just in case.
-		skips := 0
-		for range time.Tick(1 * time.Second) {
-			if skips < 600 {
-				// Check if the file changed to see if it needs to be re-read.
-				// os.Stat follows symbolic links, which is how ConfigMaps work.
-				stat, err := os.Stat(configLocation)
-				if err != nil {
-					logrus.WithField("configLocation", configLocation).WithError(err).Error("Error loading config.")
-					continue
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
 				}
-
-				recentModTime := stat.ModTime()
-
-				if !recentModTime.After(lastModTime) {
-					skips++
-					continue // file hasn't been modified
+				logrus.WithField("event", event).Info("event occurred.")
+				if event.Op&fsnotify.Write == fsnotify.Write {
```

Comment:
huh, TIL, I guess :/

Reply:
Yes. New to me too.
Copied from API example.
https://github.com/fsnotify/fsnotify/blob/master/example_test.go